include/z64actor.h

```diff
@@ -493,6 +493,29 @@ typedef enum {
     /* 0xFF */ NAVI_ENEMY_NONE = 0xFF
 } NaviEnemy;
 
+#define ACTOR_PARAMS_TRANSITION_INDEX_SHIFT 10
+#define ACTOR_TRANSITION_INDEX(actor) ((u16)(actor)->params >> ACTOR_PARAMS_TRANSITION_INDEX_SHIFT)
+
+// EnDoor and DoorKiller share openAnim and playerIsOpening
+// Due to alignment, a substruct cannot be used in the structs of these actors.
+#define DOOR_ACTOR_BASE               \
+    /* 0x0000 */ Actor actor;         \
+    /* 0x014C */ SkelAnime skelAnime; \
+    /* 0x0190 */ u8 openAnim;         \
+    /* 0x0191 */ u8 playerIsOpening;
+
+typedef struct DoorActorBase {
+    /* 0x0000 */ DOOR_ACTOR_BASE
+} DoorActorBase;
```

Comment:
am not the biggest fan of the actor struct in a macro definition, but i understand why it was done and dont have any other suggestions

Reply:
Two other possibilities I thought about:

(both untested but I think they would work)

(I think they are worse)

------------------

```c
typedef struct DoorActorBase {
    /* 0x0000 */ Actor actor;        
    /* 0x014C */ SkelAnime skelAnime;
    /* 0x0190 */ u8 openAnim;        
    /* 0x0191 */ u8 playerIsOpening;
    /* 0x0192 */ char _doorActorBaseEnd[1]; // must be the last member in this struct
} DoorActorBase;

typedef union EnDoor {
    /* 0x0000 */ DoorActorBase doorBase;
    struct {
        char _doorActorBaseBytes[offsetof(DoorActorBase, _doorActorBaseEnd)];
        /* 0x0192 */ u8 unk_192;
        /* 0x0193 */ s8 requiredObjBankIndex;
        /* 0x0194 */ s8 dListIndex;
        /* 0x0196 */ s16 lockTimer;
        /* 0x0198 */ Vec3s jointTable[5];
        /* 0x01B6 */ Vec3s morphTable[5];
        /* 0x01D4 */ EnDoorActionFunc actionFunc;
    }
} EnDoor; // size = 0x01D8
```

----------------

```c
typedef struct DoorActor {
    /* 0x0000 */ Actor actor;        
    /* 0x014C */ SkelAnime skelAnime;
    /* 0x0190 */ u8 openAnim;        
    /* 0x0191 */ u8 playerIsOpening;
    /* 0x0192 */ union {
        struct {
            /* 0x0192 */ u8 unk_192;
            /* 0x0193 */ s8 requiredObjBankIndex;
            /* 0x0194 */ s8 dListIndex;
            /* 0x0196 */ s16 lockTimer;
            /* 0x0198 */ Vec3s jointTable[5];
            /* 0x01B6 */ Vec3s morphTable[5];
            /* 0x01D4 */ EnDoorActionFunc actionFunc;
        } enDoor;
        struct {
            /* 0x0192 */ Vec3s jointTable[9];
            /* 0x01C8 */ ColliderCylinder colliderCylinder;
            /* 0x0214 */ void* texture;
            /* 0x0218 */ u16 hasHitPlayerOrGround;
            /* 0x021A */ u16 timer;
            /* 0x021C */ u8 doorObjBankIndex;
            /* 0x021D */ u8 textureEntryIndex;
            /* 0x0220 */ ColliderJntSph colliderJntSph;
            /* 0x0240 */ ColliderJntSphElement colliderJntSphItems[1];
            /* 0x0280 */ DoorKillerActionFunc actionFunc;
        } doorKiller;
    }
} DoorActor;

typedef EnDoor DoorActor;
typedef DoorKiller DoorActor;

sizeof(EnDoor) -> 0x1D8
sizeof(DoorKiller) -> 0x284
(hardcoded or define idk, or some other offsetof trick)
```

Reply:
yeah macro is still probably best in this case. some other opinions would be good